sources/Windowing/Windowing/Implementations/SDL3/SdlSurfaceComponents.OpenGL.cs

```diff
@@ -0,0 +1,431 @@
+                Sdl.GLSetAttribute(GLAttr.ContextMajorVersion, (int)ver.Major)
+                && Sdl.GLSetAttribute(GLAttr.ContextMinorVersion, (int)ver.Minor)
+            )
+        )
+        {
+            ClearErrorIf(!isInitializing);
+            return false;
+        }
+
+        // Set the bit depth.
+        if (
+            PreferredBitDepth is { } colour
+            && !(
+                (colour.X < 0 || Sdl.GLSetAttribute(GLAttr.RedSize, colour.X))
+                && (colour.Y < 0 || Sdl.GLSetAttribute(GLAttr.GreenSize, colour.Y))
+                && (colour.Z < 0 || Sdl.GLSetAttribute(GLAttr.BlueSize, colour.Z))
+                && (colour.W < 0 || Sdl.GLSetAttribute(GLAttr.AlphaSize, colour.W))
+            )
+        )
+        {
+            ClearErrorIf(!isInitializing);
+            return false;
+        }
+
+        // Configure MSAA.
+        if (
+            PreferredSampleCount is >= 0 and var count
+            && !(
+                Sdl.GLSetAttribute(GLAttr.Multisamplebuffers, 1)
+                && Sdl.GLSetAttribute(GLAttr.Multisamplesamples, count)
+            )
+        )
+        {
+            ClearErrorIf(!isInitializing);
+            return false;
+        }
+
+        // Configure depth buffer.
+        if (PreferredDepthBufferBits is { } depth && !Sdl.GLSetAttribute(GLAttr.DepthSize, depth))
+        {
+            ClearErrorIf(!isInitializing);
+            return false;
+        }
+
+        // Configure stencil buffer.
+        if (
+            PreferredStencilBufferBits is { } stencil
+            && !Sdl.GLSetAttribute(GLAttr.StencilSize, stencil)
+        )
+        {
+            ClearErrorIf(!isInitializing);
+            return false;
+        }
+
+        // Now let's try to create the context!
+        var ctx = Sdl.GLCreateContext(Handle);
+        if (ctx == nullptr)
+        {
+            ClearErrorIf(!isInitializing);
+            return false;
+        }
+
+        // If successful, we should update our state if we're initializing. If we're not initializing, then we simply
+        // destroy the context (as sadly that's the only way we can check whether a given configuration is supported).
+        if (isInitializing)
+        {
+            var swapInterval = SwapInterval;
+            ContextHandle = ctx;
+            SwapInterval = swapInterval;
+            return true;
+        }
+
+        if (!Sdl.GLMakeCurrent(currentWindow, current))
+        {
+            Sdl.ThrowError();
+        }
+
+        if (!Sdl.GLDestroyContext(ctx))
+        {
+            Sdl.ThrowError();
+        }
+
+        return true;
+        static void ClearErrorIf(bool value)
+        {
+            if (value)
+            {
+                Sdl.ClearError();
+            }
+        }
+    }
+
+    private void InitializeOpenGL(uint props)
+    {
+        if (Profile == OpenGLContextProfile.None)
+        {
+            return;
+        }
+
+        AddWindowCreateFlags(props, Sdl.WindowOpengl);
```

Comment:
What does this mean for the dummy window? Will IsSupported work as expected?

Reply:
The dummy window has now changed to be retrieved through GetDummyWindow, specifying the flags that the dummy window should have and shouldn't have. If these criteria are already met, the Handle property is returned as-is. Otherwise, the dummy window is torn down if it exists and (re)created with the proper flags.